### PR TITLE
Admin notes

### PR DESCRIPTION
**Pull request by DrSmugleaf**
## About the PR
Can be opened as verb, command or from the AHelp window.
Changes to notes are reflected across client in real time.
Notes are only visible to admins.
Any admin with the EditNotes admin flag can create, edit and delete any admin note, including those written by others.

**Screenshots**

https://user-images.githubusercontent.com/10968691/159941448-fbb8aa0d-10b8-4574-83af-7262f6fa77cb.mp4

**Changelog**
:cl:
- add: Added admin notes.

### CODE REVIEW

**Review by metalgearsloth — APPROVED** [review]

File: Content.Client/Administration/UI/Notes/AdminNotesLine.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        Separator.Visible = false;
+    }
+
+    private void Submitted(LineEditEventArgs args)
+    {
+        OnSubmitted?.Invoke(this);
+
+        AddLabel();
+
+        var note = Note with {Message = args.Text};
```
[the remark this answers is not in the document]
Comment (moonheart08) [reply Content.Client/Administration/UI/Notes/AdminNotesLine.xaml.cs:81]: apparently this is valid C# syntax in 2022. .NET 6 moment.

File: Content.Server/Administration/Notes/AdminNotesManager.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        NoteAdded?.Invoke(note);
+    }
+
+    public async Task DeleteNote(int noteId, IPlayerSession deletedBy)
+    {
+        var note = await _db.GetAdminNote(noteId);
+        if (note == null)
+        {
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (DrSmugleaf) [reply Content.Server/Administration/Notes/AdminNotesManager.cs]: Done in [this commit](https://github.com/space-wizards/space-station-14/pull/7259/commits/d2de82671851b8b9e039802f717fe57b0e95d6ba)

File: Content.Shared/Administration/AdminFlags.cs [hunk 3]
```diff
@@ -80,6 +78,16 @@ public enum AdminFlags : uint
...
 
+        /// <summary>
+        ///     Lets you view admin notes.
+        /// </summary>
+        ViewNotes = 1 << 13,
+
+        /// <summary>
+        ///     Lets you create, edit and delete admin notes.
+        /// </summary>
+        EditNotes = 1 << 14,
```
Comment (mirrorcult) [inline Content.Shared/Administration/AdminFlags.cs:89]: note to self these perms need to given to admins when merged

**Review by mirrorcult — APPROVED** [review]

**Comment by mirrorcult** [comment]
needs conflicts fixed then good to merge (?)

**Pull request merged**